packages/core/review-workflows/admin/src/routes/settings/components/WorkflowAttributes.tsx

```diff
@@ -196,5 +203,57 @@ const ContentTypeTakeNotice = styled(Typography)`
   font-style: italic;
 `;
 
+/* -------------------------------------------------------------------------------------------------
+ * StageSelector
+ * -----------------------------------------------------------------------------------------------*/
+interface StageSelectorProps {
+  disabled?: boolean;
+}
+
+const StageSelector = ({ disabled }: StageSelectorProps) => {
+  const { value: stages = [] } = useField<WorkflowStage[]>('stages');
+  const { formatMessage } = useIntl();
+
+  const { error, value, onChange } = useField('stageRequiredForPublish');
+
+  return (
+    <Field.Root
+      error={error}
+      name={'stageRequiredForPublish'}
+      hint={formatMessage({
+        id: 'settings.review-workflows.workflow.stageRequiredForPublish.hint',
+        defaultMessage: 'Prevents entries to be published if they are not at the required stage.',
+      })}
+    >
+      <Field.Label>
+        {formatMessage({
+          id: 'settings.review-workflows.workflow.stageRequiredForPublish.label',
+          defaultMessage: 'Required stage for publishing',
+        })}
+      </Field.Label>
+      <SingleSelect
+        disabled={disabled}
+        onChange={(value) => {
+          onChange('stageRequiredForPublish', value);
+        }}
+        value={value}
+      >
+        <SingleSelectOption value={''}>
+          {formatMessage({
+            id: 'settings.review-workflows.workflow.stageRequiredForPublish.any',
+            defaultMessage: 'Any stage',
+          })}
```

Comment:
Have you checked if a normal stage could also have an empty string as a name?

Reply:
Is not possible, at least at backend we have a validation check for that. While testing I discovered an error when leaving the input completely empty but not sure if it's only on this PR or is in v5. I will check and in case if on v5 I will create a PR only with that to fix it quickly 👍